### ISSUE TITLE
 Not safe goroutine for the executor

### ISSUE THREAD

**Issue by saltbo** [fission/fission#2734]
**Fission/Kubernetes version**

<pre>
$ fission version
$ kubectl version
</pre>

**Kubernetes platform (e.g. Google Kubernetes Engine)**

**Describe the bug**

process exit when the unknown panic happens in the goroutine 

**To Reproduce**

![image](https://user-images.githubusercontent.com/17308208/222616843-400c9954-3467-4bf9-a5e7-de3539d12d2c.png)

**Expected result**

recover the panic, do not exit
![image](https://user-images.githubusercontent.com/17308208/222617064-6fff5c0d-c0a0-4979-90b9-d3f7526da041.png)

**Actual result**
process exit when panic 

**Screenshots/Dump file**

<pre>
$ fission support dump
</pre>

**Additional context**

**Comment by NikhilSharmaWe** [comment]
@saltbo I think you are using the older versions. This code has been updated now.

**Comment by saltbo** [comment]
No, I use the latest versions. the panic is a test code. I want to show the process of exit when an unknown panic happens

**Comment by soharab-ic** [comment]
Please try latest fission version `v1.20.4` and re-open if issue remains.
